[Bug] Kick when trying to lay with GSit on any slab

**Describe the bug**
Someone created an issue on my AntiHealthIndicator plugin saying that when using `/lay` with the GSit plugin, he was being kicked with the error `io.netty.handler.codec.EncoderException: java.lang.NullPointerException: Cannot invoke "com.github.retrooper.packetevents.protocol.player.EquipmentSlot.getId(com.github.retrooper.packetevents.manager.server.ServerVersion)" because the return value of "com.github.retrooper.packetevents.protocol.player.Equipment.getSlot()" is null`. I have tested this, since I wanted to make sure this wasn't an issue with my plugin, and as far as I can see it isn't. I have tested this on multiple versions, and this issue does not occur on version 1.20.4 and below. 

**Software brand**
Paper 1.20.6

**Plugins**
PacketEvents, AntiHealthIndicator, [GSit](https://www.spigotmc.org/resources/gsit-modern-sit-seat-and-chair-lay-and-crawl-plugin-1-16-1-20-6.62325/)

**How To Reproduce**
1. Listen for `ENTITY_EQUIPMENT` packets
2. Make sure it is being triggered by the `/lay` command, as generally speaking it seems to work fine
3. Re-encode it

**Expected behavior**
No trace logs ;-)

**Screenshots**
```
[20:35:02 WARN]: io.netty.handler.codec.EncoderException: java.lang.NullPointerException: Cannot invoke "com.github.retrooper.packetevents.protocol.player.EquipmentSlot.getId(com.github.retrooper.packetevents.manager.server.ServerVersion)" because the return value of "com.github.retrooper.packetevents.protocol.player.Equipment.getSlot()" is null
[20:35:02 WARN]:        at io.netty.handler.codec.MessageToMessageEncoder.write(MessageToMessageEncoder.java:107)
[20:35:02 WARN]:        at packetevents-spigot-2.3.1-SNAPSHOT.jar//io.github.retrooper.packetevents.injector.handlers.PacketEventsEncoder.write(PacketEventsEncoder.java:101)
[20:35:02 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:881)
[20:35:02 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite(AbstractChannelHandlerContext.java:863)
[20:35:02 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:968)
[20:35:02 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:856)
[20:35:02 WARN]:        at io.netty.handler.codec.MessageToByteEncoder.write(MessageToByteEncoder.java:113)
[20:35:02 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:881)
[20:35:02 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite(AbstractChannelHandlerContext.java:863)
[20:35:02 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:968)
[20:35:02 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:856)
[20:35:02 WARN]:        at io.netty.handler.codec.MessageToMessageEncoder.write(MessageToMessageEncoder.java:113)
[20:35:02 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:881)
[20:35:02 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite(AbstractChannelHandlerContext.java:863)
[20:35:02 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:968)
[20:35:02 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:856)
[20:35:02 WARN]:        at io.netty.channel.ChannelOutboundHandlerAdapter.write(ChannelOutboundHandlerAdapter.java:113)
[20:35:02 WARN]:        at net.minecraft.network.Connection$2.write(Connection.java:750)
[20:35:02 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:881)
[20:35:02 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite(AbstractChannelHandlerContext.java:863)
[20:35:02 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:968)
[20:35:02 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:856)
[20:35:02 WARN]:        at io.netty.channel.ChannelDuplexHandler.write(ChannelDuplexHandler.java:115)
[20:35:02 WARN]:        at GSit-1.9.1.jar//dev.geco.gsit.mcv.v1_20_5_m.util.PackageUtil$1.write(PackageUtil.java:91)
[20:35:02 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite0(AbstractChannelHandlerContext.java:879)
[20:35:02 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.invokeWrite(AbstractChannelHandlerContext.java:863)
[20:35:02 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:968)
[20:35:02 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:856)
[20:35:02 WARN]:        at io.netty.channel.AbstractChannelHandlerContext.write(AbstractChannelHandlerContext.java:851)
[20:35:02 WARN]:        at io.netty.channel.DefaultChannelPipeline.write(DefaultChannelPipeline.java:1010)
[20:35:02 WARN]:        at io.netty.channel.AbstractChannel.write(AbstractChannel.java:296)
[20:35:02 WARN]:        at net.minecraft.network.Connection.doSendPacket(Connection.java:505)
[20:35:02 WARN]:        at net.minecraft.network.Connection.lambda$sendPacket$13(Connection.java:490)
[20:35:02 WARN]:        at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174)
[20:35:02 WARN]:        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167)
[20:35:02 WARN]:        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
[20:35:02 WARN]:        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:569)
[20:35:02 WARN]:        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
[20:35:02 WARN]:        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
[20:35:02 WARN]:        at java.base/java.lang.Thread.run(Thread.java:1583)
[20:35:02 WARN]: Caused by: java.lang.NullPointerException: Cannot invoke "com.github.retrooper.packetevents.protocol.player.EquipmentSlot.getId(com.github.retrooper.packetevents.manager.server.ServerVersion)" because the return value of "com.github.retrooper.packetevents.protocol.player.Equipment.getSlot()" is null
[20:35:02 WARN]:        at packetevents-spigot-2.3.1-SNAPSHOT.jar//com.github.retrooper.packetevents.wrapper.play.server.WrapperPlayServerEntityEquipment.write(WrapperPlayServerEntityEquipment.java:84)
[20:35:02 WARN]:        at packetevents-spigot-2.3.1-SNAPSHOT.jar//com.github.retrooper.packetevents.util.PacketEventsImplHelper.handleClientBoundPacket(PacketEventsImplHelper.java:53)
[20:35:02 WARN]:        at packetevents-spigot-2.3.1-SNAPSHOT.jar//io.github.retrooper.packetevents.injector.handlers.PacketEventsEncoder.handleClientBoundPacket(PacketEventsEncoder.java:82)
[20:35:02 WARN]:        at packetevents-spigot-2.3.1-SNAPSHOT.jar//io.github.retrooper.packetevents.injector.handlers.PacketEventsEncoder.encode(PacketEventsEncoder.java:67)
[20:35:02 WARN]:        at packetevents-spigot-2.3.1-SNAPSHOT.jar//io.github.retrooper.packetevents.injector.handlers.PacketEventsEncoder.encode(PacketEventsEncoder.java:46)
[20:35:02 WARN]:        at io.netty.handler.codec.MessageToMessageEncoder.write(MessageToMessageEncoder.java:90)
[20:35:02 WARN]:        ... 39 more
```

**Additional context**
This only occurs when running the `/lay` command on 1.20.5+. The above stacktrace is from Booky's custom build to add more details around an warning / error.

--- a/api/src/main/java/com/github/retrooper/packetevents/protocol/player/EquipmentSlot.java
+++ b/api/src/main/java/com/github/retrooper/packetevents/protocol/player/EquipmentSlot.java
@@ -27,7 +27,8 @@ public enum EquipmentSlot {
     BOOTS(1),
     LEGGINGS(2),
     CHEST_PLATE(3),
-    HELMET(4);
+    HELMET(4),
+    BODY(0);
 
     private static final EquipmentSlot[] VALUES = values();
 

--- a/api/src/main/java/com/github/retrooper/packetevents/protocol/player/EquipmentSlot.java
+++ b/api/src/main/java/com/github/retrooper/packetevents/protocol/player/EquipmentSlot.java
@@ -39,7 +39,7 @@ public enum EquipmentSlot {
     }
 
     public int getId(ServerVersion version) {
-        return getId(serverVersion.toClientVersion());
+        return getId(version.toClientVersion());
     }
 
     public int getId(ClientVersion version) {

--- a/api/src/main/java/com/github/retrooper/packetevents/protocol/player/EquipmentSlot.java
+++ b/api/src/main/java/com/github/retrooper/packetevents/protocol/player/EquipmentSlot.java
@@ -35,26 +35,37 @@ public enum EquipmentSlot {
     private final byte legacyId;
 
     EquipmentSlot(int legacyId) {
-        this.legacyId =(byte) legacyId;
+        this.legacyId = (byte) legacyId;
     }
 
     public int getId(ServerVersion version) {
-        if (version.isOlderThan(ServerVersion.V_1_9)) {
+        return getId(serverVersion.toClientVersion());
+    }
+
+    public int getId(ClientVersion version) {
+        if (version.isOlderThan(ClientVersion.V_1_9)) {
             return legacyId;
-        }
-        else {
+        } else {
             return ordinal();
         }
     }
 
     @Nullable
-    public static EquipmentSlot getById(ServerVersion version, int id) {
-        // FIXME: try making this O(1)
-        for (EquipmentSlot slot : VALUES) {
-            if (slot.getId(version) == id) {
-                return slot;
+    public static EquipmentSlot getById(ClientVersion version, int id) {
+        if (version.isOlderThan(ClientVersion.V_1_9)) {
+            for (EquipmentSlot slot : VALUES) {
+                if (slot.getId(version) == id) {
+                    return slot;
+                }
             }
+        } else {
+            return VALUES[id];
         }
         return null;
+    }
+
+    @Nullable
+    public static EquipmentSlot getById(ServerVersion version, int id) {
+        return getById(version.toClientVersion(), id);
     }
 }

